repose-aggregator/components/filters/ip-user-filter/src/main/scala/org/openrepose/filters/ipuser/IpUserFilter.scala

```diff
@@ -77,16 +77,21 @@ class IpUserFilter @Inject()(configurationService: ConfigurationService) extends
       val clientIpAddress = request.getSplittableHeaderScala(CommonHttpHeader.X_FORWARDED_FOR)
         .headOption.getOrElse(servletRequest.getRemoteAddr)
 
-      getClassificationLabel(clientIpAddress).foreach { label =>
-        request.addHeader(groupHeaderName, label, groupHeaderQuality)
-      }
+      try {
+        getClassificationLabel(clientIpAddress).foreach { label =>
+          request.addHeader(groupHeaderName, label, groupHeaderQuality)
+        }
 
-      //Always set the user header name to the current IP address
-      request.addHeader(userHeaderName, clientIpAddress, userHeaderQuality)
-
-      logger.trace("IP User filter passing request...")
-      filterChain.doFilter(request, servletResponse)
+        //Always set the user header name to the current IP address
+        request.addHeader(userHeaderName, clientIpAddress, userHeaderQuality)
 
+        logger.trace("IP User filter passing request...")
+        filterChain.doFilter(request, servletResponse)
+      } catch {
+        case uhe: UnknownHostException =>
+          logger.debug(s"Bad X-Forwarded-For header: $clientIpAddress", uhe)
```

Comment:
Technically, this exception could be thrown either by a bad `X-Forwarded-For` header or by a bad value returned by `getRemoteAddr`.

Reply:
Yes, but I don't see how we'd get a bad value out of `getRemoteAddr`

Reply:
Oh, it's quite easy to get a bad value since the return value is just a `String`. I think my example below is the easiest to see; an improperly implemented request wrapper could, say, return `/` which would lead to this exception being thrown.

Of course, we _should not_ get a bad value from `getRemoteAddr`, but since Repose is extensible and operators can inject their own code we _could_ get a bad value from `getRemoteAddr`.

So it's a bit of defensive coding to account for that, and stating expected behavior if that were to happen.

Reply:
Yes, a bad wrapper will do it, but it will never be because of the user. And if we really want to cover that it's much more complicated. we have to do our own tests at different times so we know which one we were looking at and which one failed because because the behavior should be a 400 for the header check, but 500 for the remote address.

Reply:
To be clear, I felt the likelihood of the other case was not worth the added complexity.

Reply:
Fair enough. Something about conflating the sources of the IP address bothers me, but I see how accounting for those sources adds complexity for an unlikely case.